audio/src/passthrough_decoder.rs

```diff
@@ -5,75 +5,32 @@ use std::fmt;
 use std::io::{Read, Seek};
 use std::time::{SystemTime, UNIX_EPOCH};
 
-fn write_headers<T: Read + Seek>(
-    rdr: &mut PacketReader<T>,
-    wtr: &mut PacketWriter<Vec<u8>>,
-) -> Result<u32, PassthroughError> {
-    let mut stream_serial: u32 = SystemTime::now()
-        .duration_since(UNIX_EPOCH)
-        .unwrap()
-        .as_millis() as u32;
-
-    // search for ident, comment, setup
-    get_header(1, rdr, wtr, &mut stream_serial, PacketWriteEndInfo::EndPage)?;
-    get_header(
-        3,
-        rdr,
-        wtr,
-        &mut stream_serial,
-        PacketWriteEndInfo::NormalPacket,
-    )?;
-    get_header(5, rdr, wtr, &mut stream_serial, PacketWriteEndInfo::EndPage)?;
-
-    // remove un-needed packets
-    rdr.delete_unread_packets();
-    return Ok(stream_serial);
-}
-
-fn get_header<T>(
-    code: u8,
-    rdr: &mut PacketReader<T>,
-    wtr: &mut PacketWriter<Vec<u8>>,
-    stream_serial: &mut u32,
-    info: PacketWriteEndInfo,
-) -> Result<u32, PassthroughError>
+fn get_header<T>(code: u8, rdr: &mut PacketReader<T>) -> Result<Box<[u8]>, PassthroughError>
 where
     T: Read + Seek,
 {
     let pck: Packet = rdr.read_packet_expected()?;
 
-    // set a unique serial number
-    if pck.stream_serial() != 0 {
-        *stream_serial = pck.stream_serial();
-    }
-
     let pkt_type = pck.data[0];
     debug!("Vorbis header type{}", &pkt_type);
 
-    // all headers are mandatory
     if pkt_type != code {
         return Err(PassthroughError(OggReadError::InvalidData));
     }
 
-    // headers keep original granule number
-    let absgp_page = pck.absgp_page();
-    wtr.write_packet(
-        pck.data.into_boxed_slice(),
-        *stream_serial,
-        info,
-        absgp_page,
-    )
-    .unwrap();
-
-    return Ok(*stream_serial);
+    return Ok(pck.data.into_boxed_slice());
 }
 
 pub struct PassthroughDecoder<R: Read + Seek> {
     rdr: PacketReader<R>,
     wtr: PacketWriter<Vec<u8>>,
-    lastgp_page: Option<u64>,
-    absgp_page: u64,
+    eos: bool,
+    bos: bool,
+    ofsgp_page: u64,
```

Comment:
I never understood why C programmers are afraid of using many letters for variables.

Reply:
ah, I started a long time ago to be a DSP programmer as well as doing X86 assembly code. C was a luxury and you had to count every byte, so I kept the "frugal" mindset from these days. Having said that, it's also because the Rust formater shrinks everything to 80 columns and that, I find it absolutely crazy in 2021. I think these days are long gone and even Linux Torvald :-) thinks that should be changed. So that 80 column pushed the habit of short variable names because having 10 lines of text for one line of code is ugly, IMHO. Last but not least, I'm always trying to fit with the style of the code I'm adding my changes to and this started from the "ogg/vorbis" library that uses that same short naming for offset and granule.